commitchecker/README.md

```diff
@@ -0,0 +1,18 @@
+# commitchecker
+
+commitchecker validates a range of commits in a git repository and ensures they meet specific requirements:
+
+1. The author's email address does not start with "root@".
+2. The message starts with one of:
+   1. UPSTREAM: <PR number|carry|drop>: description
+   2. UPSTREAM: revert: <normal upstream format>
+
+This is useful for repositories that are downstream forks of upstream repositories.
+
+# History
+
+This comes from
+from https://github.com/openshift/kubernetes/tree/0abcd84431df81ed9f2a1846b5045e46d9032cc1/openshift-hack/commitchecker.
+
+That repository can't be vendored, and you can't use `go get` against it (for multiple reasons), so we extracted the
```

Comment:
I'm curious why this is.

Reply:
openshift/kubernetes is a fork of kubernetes/kubernetes. We didn't change the [go module](https://github.com/openshift/kubernetes/blob/master/go.mod#L7) from k8s.io/kubernetes to github.com/openshift/kubernetes (nor should we). That means if you try to `go get github.com/openshift/kubernetes`, it will fail because of the module mismatch.

Additionally, the git history in the openshift/kubernetes repo is huge, and it takes a long time (and bandwidth) to clone it. Some tools, such as [bingo](https://github.com/bwplotka/bingo), time out trying to retrieve the repo.

It makes more sense IMO to move this tool into a more appropriate repo, such as this one, to allow easier usage by other repos wanting to use it.

Reply:
Ack, thanks for the explanation :+1: